src/web/services/http-request.service.ts

```diff
@@ -1,10 +1,35 @@
-import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { HttpClient, HttpHeaders, HttpParameterCodec, HttpParams } from '@angular/common/http';
 import { Injectable } from '@angular/core';
 import { Observable } from 'rxjs';
 import { environment } from '../environments/environment';
 import { ResourceEndpoints } from '../types/api-endpoints';
 import { MasqueradeModeService } from './masquerade-mode.service';
 
+/**
+ * This encoder was created to support encoding and decoding of plus (+) signs.
+ *
+ * Angular will ignore the encoding for plus signs. Refer to:
+ * https://github.com/angular/angular/blob/8.0.0/packages/common/http/src/params.ts#L33
+ */
+class CustomEncoder implements HttpParameterCodec {
+
+  encodeKey(key: string): string {
+    return encodeURIComponent(key);
```

Comment:
https://github.com/angular/angular/blob/b72fce8acf01639438df5ef92473e088f35e2e94/packages/common/http/src/params.ts#L81-L92

There should be a reason why they do it like this. Should just copy that and modify accordingly (if my hunch is right, remove line 88).

Reply:
Do we want to remove the slash as well? i.e. is there a list of permissible special characters for session names?